tests/tel2puml/find_unique_graphs/test_otel_data_holder.py

```diff
@@ -0,0 +1,334 @@
+                assert node_0.event_id == "0"
+                assert node_0.start_timestamp == 1695639486119918080
+                assert node_0.end_timestamp == 1695639486119918084
+                assert node_0.application_name == "test_application_name"
+                assert node_0.parent_event_id == "None"
+
+                # Test parent-child relationship
+                node_1 = (
+                    session.query(NodeModel).filter_by(event_id="1").first()
+                )
+                assert node_0.children == [node_1]
+            else:
+                raise TypeError(
+                    "node_0 should be of type 'NodeModel',"
+                    f" got '{type(node_0)}' instead."
+                )
+
+    @staticmethod
+    def test_commit_batched_data_success(
+        mock_sql_config: SQLDataHolderConfig,
+    ) -> None:
+        """Tests commit_batched_data_to_database method, with success."""
+
+        holder = SQLDataHolder(mock_sql_config)
+        parent_node_model = NodeModel(
+            job_name="test_job_name",
+            job_id="test_job_id",
+            event_type="test_event",
+            event_id="100",
+            start_timestamp=1723544154817893024,
+            end_timestamp=1723544154817798024,
+            application_name="test_app",
+            parent_event_id=None,
+        )
+        child_node_model = NodeModel(
+            job_name="test_job_name",
+            job_id="test_job_id",
+            event_type="test_event",
+            event_id="101",
+            start_timestamp=1723544154817893024,
+            end_timestamp=1723544154817798024,
+            application_name="test_app",
+            parent_event_id="100",
+        )
+
+        holder.node_relationships_to_save.append(
+            {"parent_id": "100", "child_id": "101"}
+        )
+        holder.node_models_to_save.append(parent_node_model)
+        holder.node_models_to_save.append(child_node_model)
+
+        # Test reset batch
+        assert len(holder.node_models_to_save) == 2
+        assert len(holder.node_relationships_to_save) == 1
+
+        holder.commit_batched_data_to_database()
+
+        assert len(holder.node_models_to_save) == 0
+        assert len(holder.node_relationships_to_save) == 0
+
+        with holder.session as session:
+            # Test NodeModels are correctly stored in the database
+            node_0 = session.query(NodeModel).filter_by(event_id="100").first()
+            node_1 = session.query(NodeModel).filter_by(event_id="101").first()
+
+            if isinstance(node_0, NodeModel):
+                assert node_0.job_name == "test_job_name"
+            else:
+                raise TypeError(
+                    "node_0 should be of type 'NodeModel',"
+                    f" got '{type(node_0)}' instead."
+                )
+
+            if isinstance(node_1, NodeModel):
+                assert node_1 in node_0.children
+                assert node_1.application_name == "test_app"
```

Comment:
would prefer if you checked against all the attributes to make sure they're the same. Would just loop over a list of attributes and make sure they are equal i.e. 
```
for attr in ["job_name", etc]:
    assert getattr(node0, attr) == getattr(parent_node, attr)
   etc

Reply:
This wasn't resolved